From 6ea6004073f3642046bc7b677576d1565d0bf5a5 Mon Sep 17 00:00:00 2001
From: Axel Huebl <axel.huebl@plasma.ninja>
Date: Wed, 2 Oct 2024 12:40:29 -0700
Subject: [PATCH 1/2] Fix: `newVersion.sh` Python

The script was not robust enough for the black/ruff
formatting updates we applied. This affected the 24.09
and 24.10 release tags.

This fixes it.
---
 Tools/Release/newVersion.sh | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/Tools/Release/newVersion.sh b/Tools/Release/newVersion.sh
index b1d2a6aad27..9491401b120 100755
--- a/Tools/Release/newVersion.sh
+++ b/Tools/Release/newVersion.sh
@@ -104,25 +104,25 @@ sed -i -E "s/"\
 
 # setup.py: version = '21.02',
 sed -i -E "s/"\
-"([[:blank:]]*version[[:blank:]]*=[[:blank:]]*')(.*)('.+)/"\
+"([[:blank:]]*version[[:blank:]]*=[[:blank:]]*\")(.*)(\".+)/"\
 "\1${VERSION_STR}\3/g" \
     ${REPO_DIR}/setup.py
 
 # Python/setup.py: version = '21.02',
 sed -i -E "s/"\
-"([[:blank:]]*version[[:blank:]]*=[[:blank:]]*')(.*)('.+)/"\
+"([[:blank:]]*version[[:blank:]]*=[[:blank:]]*\")(.*)(\".+)/"\
 "\1${VERSION_STR}\3/g" \
     ${REPO_DIR}/Python/setup.py
 
 # sphinx / RTD
 #   docs/source/conf.py
 sed -i "s/"\
-"[[:blank:]]*version[[:blank:]]*=[[:blank:]]*u.*/"\
-"version = u'${VERSION_STR_NOSUFFIX}'/g" \
+"[[:blank:]]*version[[:blank:]]*=[[:blank:]]*.*/"\
+"version = \"${VERSION_STR_NOSUFFIX}\"/g" \
     ${REPO_DIR}/Docs/source/conf.py
 sed -i "s/"\
-"[[:blank:]]*release[[:blank:]]*=[[:blank:]]*u.*/"\
-"release = u'${VERSION_STR}'/g" \
+"[[:blank:]]*release[[:blank:]]*=[[:blank:]]*.*/"\
+"release = \"${VERSION_STR}\"/g" \
     ${REPO_DIR}/Docs/source/conf.py
 
 

From 3e6b0b85486639d80fec9e6af776a0836b68c405 Mon Sep 17 00:00:00 2001
From: Axel Huebl <axel.huebl@plasma.ninja>
Date: Wed, 2 Oct 2024 12:41:19 -0700
Subject: [PATCH 2/2] Python: 24.10 Versions

Update missing versions using the revised script.
---
 Docs/source/conf.py | 4 ++--
 Python/setup.py     | 2 +-
 setup.py            | 2 +-
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/Docs/source/conf.py b/Docs/source/conf.py
index 9dfda6346f9..c1ad43197c5 100644
--- a/Docs/source/conf.py
+++ b/Docs/source/conf.py
@@ -107,9 +107,9 @@ def __init__(self, *args, **kwargs):
 # built documents.
 #
 # The short X.Y version.
-version = "24.08"
+version = "24.10"
 # The full version, including alpha/beta/rc tags.
-release = "24.08"
+release = "24.10"
 
 # The language for content autogenerated by Sphinx. Refer to documentation
 # for a list of supported languages.
diff --git a/Python/setup.py b/Python/setup.py
index 86585bf8886..d57ebc65223 100644
--- a/Python/setup.py
+++ b/Python/setup.py
@@ -65,7 +65,7 @@
 
 setup(
     name="pywarpx",
-    version="24.08",
+    version="24.10",
     packages=["pywarpx"],
     package_dir={"pywarpx": "pywarpx"},
     description="""Wrapper of WarpX""",
diff --git a/setup.py b/setup.py
index efc18d900cf..9683c8ab675 100644
--- a/setup.py
+++ b/setup.py
@@ -274,7 +274,7 @@ def build_extension(self, ext):
 setup(
     name="pywarpx",
     # note PEP-440 syntax: x.y.zaN but x.y.z.devN
-    version="24.08",
+    version="24.10",
     packages=["pywarpx"],
     package_dir={"pywarpx": "Python/pywarpx"},
     author="Jean-Luc Vay, David P. Grote, Maxence Thévenet, Rémi Lehe, Andrew Myers, Weiqun Zhang, Axel Huebl, et al.",
